chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java

```diff
@@ -112,30 +113,30 @@ public Builder withInvoker(Class<? extends Invoker> invoker) {
         }
 
         /**
-         * Optional: Adds a list of subtypes of {@link com.griddynamics.jagger.engine.e1.collector.ResponseValidator}
+         * Optional: Adds a list of subtypes of {@link com.griddynamics.jagger.engine.e1.scenario.KernelSideObjectProvider<ResponseValidator>}
          * Instances of those subtypes will be used to validate responses during Jagger test execution @n
          * Example:
          * @code
-         *      addValidators(Arrays.asList(com.griddynamics.jagger.engine.e1.collector.NotNullResponseValidator.class))
+         *      addValidator(ResponseValidatorProvider.of(NotNullResponseValidator.class))
          * @endcode
          * @see com.griddynamics.jagger.engine.e1.collector.NotNullResponseValidator for example
          */
-        public Builder addValidators(List<Class<? extends ResponseValidator>> validators) {
+        public Builder addValidators(List<KernelSideObjectProvider<ResponseValidator<Object, Object, Object>>> validators) {
```

Comment:
What about to remove <Object, Object, Object> and make signature of this method like addListeners.
Also I propose create interface like
`interface ValidatorProvider<Q, E, R> implements KernelSideObjectProvider<ResponseValidator<Q, E, R>>`
to make code more readable and user friendly %) 
And add some example with custom provider, for example based on TrueValidator